app/views/homepage/_grid_item.haml

```diff
@@ -9,5 +9,15 @@
         .home-fluid-thumbnail-grid-author-avatar
           = link_to(person_path(id: listing.author.username)) do
             = image_tag(listing.author.avatar.thumb || missing_avatar(:thumb), :class => "home-fluid-thumbnail-grid-author-avatar-image")
-        = link_to(person_path(id: listing.author.username), :class => "home-fluid-thumbnail-grid-author-name") do
-          = PersonViewUtils::person_entity_display_name(listing.author, @current_community.name_display_type)
+
+        - distance = Maybe(listing.distance).or_else(nil)
+        - if(!show_distance || distance.blank?)
+          = link_to(person_path(id: listing.author.username), :class => "home-fluid-thumbnail-grid-author-name") do
+            = PersonViewUtils::person_entity_display_name(listing.author, @current_community.name_display_type)
+        - else
+          .home-fluid-thumbnail-grid-details
+            = link_to(person_path(id: listing.author.username), class: "home-fluid-thumbnail-grid-details-author-name") do
+              = PersonViewUtils::person_entity_display_name(listing.author, @current_community.name_display_type)
+            .home-fluid-thumbnail-grid-details-distance
+              - dist = number_with_delimiter(distance.round(2))
```

Comment:
number_with_delimiter defaults to locale, but having it visible could make this more understandable. I'll change it.
´´´
number_with_delimiter(distance.round(2), locale: locale)
´´´